src/icon2xml.rs

```diff
@@ -0,0 +1,121 @@
+//! Produces Android Vector Drawable XML of icons in Google-style icon fonts
+
+use crate::{
+    error::DrawSvgError, icon2svg::DrawOptions, pathstyle::SvgPathStyle, pens::SvgPathPen,
+};
+use kurbo::Affine;
+use skrifa::{
+    instance::Size as SkrifaSize, outline::pen::PathStyle, outline::DrawSettings,
+    raw::TableProvider, FontRef, MetadataProvider,
+};
+
+pub fn draw_xml(font: &FontRef, options: &DrawOptions<'_>) -> Result<String, DrawSvgError> {
+    let upem = font
+        .head()
+        .map_err(|e| DrawSvgError::ReadError("head", e))?
+        .units_per_em();
+    let gid = options
+        .identifier
+        .resolve(font, &options.location)
+        .map_err(|e| DrawSvgError::ResolutionError(options.identifier.clone(), e))?;
+
+    let glyph = font
+        .outline_glyphs()
+        .get(gid)
+        .ok_or(DrawSvgError::NoOutline(options.identifier.clone(), gid))?;
+
+    let (viewport_width, viewport_height) = if options.use_width_height_for_viewbox {
+        (options.width_height, options.width_height)
+    } else {
+        (upem as f32, upem as f32)
+    };
+    let scale = viewport_width as f64 / upem as f64;
+
+    // Fonts are Y-up, vector drawable Y-down.
+    // So we need to flip y and translate.
+    // y' = viewport_height - y_glyph * scale
+    // x' = x_glyph * scale
+    let transform = Affine::new([scale, 0.0, 0.0, -scale, 0.0, viewport_height as f64]);
+    let mut pen = SvgPathPen::new_with_transform(transform);
+
+    glyph
+        .draw(
+            DrawSettings::unhinted(SkrifaSize::unscaled(), options.location)
+                .with_path_style(PathStyle::HarfBuzz),
+            &mut pen,
+        )
+        .map_err(|e| DrawSvgError::DrawError(options.identifier.clone(), gid, e))?;
```

Comment:
isn't this duplicated as well, possible exception of how you computed the transform?